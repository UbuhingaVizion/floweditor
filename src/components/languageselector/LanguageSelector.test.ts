import {
  containerClasses,
  LanguageSelector,
  LanguageSelectorProps
} from 'components/languageselector/LanguageSelector';
import { composeComponentTestUtils, setMock, setupStore } from 'testUtils';
import { English, languages, Spanish } from 'testUtils/assetCreators';

const baseProps: LanguageSelectorProps = {
  language: English,
  languages,
  handleLanguageChange: jest.fn()
};

const { setup } = composeComponentTestUtils(LanguageSelector, baseProps);

describe(LanguageSelector.name, () => {
  it('renders', () => {
    setupStore({ languageCode: 'spa' });
    const { wrapper } = setup(true);
    expect(wrapper).toMatchSnapshot();
  });
});
