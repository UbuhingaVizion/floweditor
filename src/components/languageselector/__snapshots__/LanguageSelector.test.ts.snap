// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSelector renders 1`] = `
<div
  className="language-selector"
>
  <div
    style={
      Object {
        "marginTop": "2em",
        "position": "fixed",
        "right": "2em",
        "zIndex": 1,
      }
    }
  >
    <div
      className="language"
      key="lang_eng"
    >
      <div
        className="language_link "
        onClick={[Function]}
      >
        English
      </div>
    </div>
    <div
      className="language"
      key="lang_spa"
    >
      <div
        className="separator"
      >
        |
      </div>
      <div
        className="language_link active"
        onClick={[Function]}
      >
        Spanish
      </div>
    </div>
  </div>
</div>
`;
